From 5ea2f8709b4d091700750661231f8a3ddce0fc7c Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Thu, 10 Dec 2020 16:00:50 +0100
Subject: [PATCH] Fix PyTuple_SET_ITEM() usage - no return value

As noted in bpo-30459 (link bellow) the PyTuple_SET_ITEM() macro
has not a return value. Let's make code compatible with python 3.10.

Link: https://bugs.python.org/issue30459
BugLink: https://github.com/alsa-project/alsa-python/issues/2
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 pyalsa/alsahcontrol.c | 4 ++--
 pyalsa/alsamixer.c    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/pyalsa/alsahcontrol.c b/pyalsa/alsahcontrol.c
index ebee1b7..7c9321f 100644
--- a/pyalsa/alsahcontrol.c
+++ b/pyalsa/alsahcontrol.c
@@ -1543,8 +1543,8 @@ static int element_callback(snd_hctl_elem_t *elem, unsigned int mask)
 
 	t = PyTuple_New(2);
 	if (t) {
-		if (PyTuple_SET_ITEM(t, 0, (PyObject *)pyhelem))
-			Py_INCREF(pyhelem);
+		PyTuple_SET_ITEM(t, 0, (PyObject *)pyhelem);
+		Py_INCREF(pyhelem);
 		PyTuple_SET_ITEM(t, 1, PyInt_FromLong(mask));
 		r = PyObject_CallObject(o, t);
 		Py_DECREF(t);
diff --git a/pyalsa/alsamixer.c b/pyalsa/alsamixer.c
index 39d7387..91fe198 100644
--- a/pyalsa/alsamixer.c
+++ b/pyalsa/alsamixer.c
@@ -1348,8 +1348,8 @@ static int element_callback(snd_mixer_elem_t *elem, unsigned int mask)
 
 	t = PyTuple_New(2);
 	if (t) {
-		if (PyTuple_SET_ITEM(t, 0, (PyObject *)pyelem))
-			Py_INCREF(pyelem);
+		PyTuple_SET_ITEM(t, 0, (PyObject *)pyelem);
+		Py_INCREF(pyelem);
 		PyTuple_SET_ITEM(t, 1, PyInt_FromLong(mask));
 		r = PyObject_CallObject(o, t);
 		Py_DECREF(t);
